Source/Core/CesiumTerrainProvider.js

```diff
@@ -840,9 +1015,93 @@ define([
         if (!defined(this._availability)) {
             return undefined;
         }
+        if (level > this._availability._maximumLevel) {
+            return false;
+        }
 
-        return this._availability.isTileAvailable(level, x, y);
+        var result = loadTileAvailability(this, x, y, level);
+
+        // If we got a result, return it. If it was a promise return false for now.
```

Comment:
In the `fill-tiles` branch, returning false here will cause the terrain engine to upsample the tile instead of loading it, and never look back. We should return some sort of sentinel value (null? unfortunately we're already using undefined) to indicate that we don't know yet, rather than returning false.